clarify election process for operations advisory group (#20)
solid/odi-governance: README.md

As ODI gets resources to start supporting the implementation of a roadmap for Solid, and to better help coordinate the community at a technical level on tasks such as codebase creation and maintenance \- this advisory group would also feed back into the roadmap prioritisation.

> The Solid Operations Advisory Group consists of three members elected using the [ranked choice voting process used to select Chairs of the Solid Community Group](https://www.w3.org/community/solid/charter/). Members are elected for 2 year terms. Specifically,

Author: ```suggestion
The Solid Operations Advisory Group consists of three members elected using a Broda count voting process. Members are elected for 2 year terms. Specifically,
```

>  - There is a self-nomination process. Candidates are asked to submit a statement justifying their suitability to the role - including experiences and background that means they will ensure accessibility of infrastructure to a diverse range of developers and end users - towards the global vision of Solid. This will be made via a PR to a specified document in this repository. Nominees will have **10 days** from the date of announcement to self-nominate.

Author: @jeff-zucker @CxRes @mrkvon - could you please identify if you would like any modifications to the phrasing of the self-nomination statement

Reviewer: @jeswr You need to setup a small hoop for nomination process. This could be self-nomination, if you have previously contributed to Solid or requiring, say, two other community members (or previous solid contributors) to sign off on your nomination. This is a basic sanity check to prevent random applications.

Also the ranked choice voting for a committee is fundamentally flawed, as it selects from everyone's top choice (adjusted for eliminations) to fill n seats as opposed to everyone's top n choices to fill n seats. This is why, for example, elf Pavlik did not get elected in the first round of CG elections as he was a 2nd choice for a lot of people voting strategically. I would suggest Broda Count but that's my preference.

Author: Good thought - thinking out loud perhaps the phrasing could be:

Must have meaningully contributed to one of the following within the last 12 months, or be endorsed by two community members who have have meaningully contributed to one of the following within the last 12 months:
 - solidproject.org
 - solidcommunity.net
 - Any repositories under the [solid](https://github.com/solid/) and [solid-contrib](https://github.com/solid-contrib/) GitHub organisations.
 
Got it on ranked choice being flawed - have you researched to see if the Broda Count is the best alternative / what is the suggestion based on?

Reviewer 2: The Solid CG used a Single Transferable Vote system because like with most of its charter/guidelines leaning on the W3C Process where applicable, it was one of them for various elections at W3C (AB, TAG, at the very least): https://www.w3.org/policies/process/#AB-TAG-elections . And, neither the CG or myself had the time or the expertise to figure out the most suitable voting method *in combination with the voting system* (provided by W3C) we could use (with some software we had to write tapping into W3C API for eligibility etc etc..). Perfect/Flawless? Most likely not. Most suitable considering various criteria/options? I would say so.

And to be more specific, W3C uses [MeekSTV](https://en.wikipedia.org/wiki/Counting_single_transferable_votes#Meek).

Regarding who got elected and who not.. there is the [Solid CG Election Retrospective](https://github.com/solid/specification/blob/main/meetings/2024-01-09.md#solid-cg-election-retrospective) which might be of some use. 

>SC: I don't know how to evaluate the results in a meaningful and fair 
way without understanding MeekSTV well. I'll trust the consensus in W3C 
as to why they use it and keep it aligned to rest of election stuff 
there. Easier to work with a process that they have than doing our own 
things, in case we have questions, etc.

>SC: That doesn't mean we cannot do additional things on top of things 
we observe. Charter has some constraints on only one person from each 
affiliation; we can add other constraints on other things. I don't know 
if there should be some thresholds. Comes down to the math. Looking at 
these results: one of the winners has 39% unranked. I don't know how to 
interpret that, but at face value, from 41 ballots, 39% said they did 
not want the candidate, but they got a seat. I don't know what to do 
with that information, but it does stand out. It had to do with the method.

---

So, I don't have a strong opinion on a particular STV - folks with more time, expertise, and community support/history would be better fit to speak about it - but generally find it fair. Except of course when we get strange outcomes such as the one above - while 39% of the voters not even ranking a candidate (~do not want) yet they still get a seat.

Reviewer: > Got it on ranked choice being flawed - have you researched to see if the Broda Count is the best alternative / what is the suggestion based on?

Let me address that in two parts:

1. A perfect voting system is mathematically impossible. See for example: <https://www.youtube.com/watch?v=qf7ws2DF-zk>
2. There is some research that favours Broda count. See a summary here: <https://marginalrevolution.com/marginalrevolution/2025/01/the-borda-count-is-the-best-method-of-voting.html> but also see counterpoints in the comments.

> And, neither the CG or myself had the time or the expertise to figure out the most suitable voting method in combination with the voting system (provided by W3C) 

Like @csarven points out, this all came to us in hindsight after the Solid CG election. However, up to a first approximation, I am reasonably confident of my reasoning of the results.  I think the W3C might be well advised to re-examine its choice of voting system, particularly for multi-candidate elections.

Reviewer: > Must have meaningully contributed ...

This is OK as long as there are clear guidelines on what constitutes "meaningful contribution", and an authority to decide this, perhaps someone from the AC.

Reviewer 2: I've shared some findings with W3C on the AC forum (member confidential): https://lists.w3.org/Archives/Member/w3c-ac-forum/2024OctDec/0107.html

re meaningfully contributed, that may be tough to sketch out but I'd be curious to know what that may be :) Perhaps equally significant is being in a good standing.

Reviewer 3: I'm generally in favor of a CG-like voting process although I don't believe that it should be tied to CG membership.  Participation in other Solid related groups such as the Practitioners, Data-Modules Group, SolidOS Group, Solid Symposium, etc. should also qualify someone as able to vote.  If we go with requiring CG membership, someone should look into the problems experienced by some users who were not able to complete the application for W3C membership because they had an employer but were seeking to join as an unaffiliated individual rather than as a representative of their employer (who had nothing to do with Solid).

Author: @jeff-zucker @CxRes @csarven @pchampin 

Thank you for all of your explanations - I see the issues with tying to CG membership (I am especially shocked by the challenges of joining). An option is to use the same hoop that @CxRes proposed and I subsequently defined in https://github.com/solid/odi-governance/pull/20#discussion_r1987183771 to define both the self-nominee and voting pool - however, I fear this is too restrictive. On the other end of the spectrum we could make it an open vote - providing we find a non-exclusionary mechanism to authenticate people to prevent duplicate/spam voting and prevent multiple votes from the same organisations. Do you have a preference towards either of these two approaches - or an alternative?

Author: As far as meaningful contributions - I had written this with the goal of protecting against people making trivial changes just to get on the OAC (e.g. adding a full stop to some docs 20min before submitting a vote).

If that as the understood intention, I think this should be fairly objective to judge - the question to me is whether it is a high enough bar to self-nominate.

Reviewer 4: "meaningful contributions" seems too restrictive for voters, but I don't have anything better to propose :-/

Reviewer: I realize this as well, but options I can think of are silly!

Author: Thanks for all of the inputs, based on this thread I have:
 - Changed to using a Broda count, my understanding is that there were no objections to this choice
 - Removed any membership barriers to vote
 
See changes in https://github.com/solid/odi-governance/pull/20/commits/c2c7a76869248706df4052bfbc534cccfc6f44a6

Author: I'm going to merge this PR - should further changes be needed please open an issue.

Reviewer 5: Drugs

>  - There will be a **10 day** period in which to vote for nominees - to vote you must be a member of the Solid Community Group and the example [ranked choice voting process used to select Chairs of the Solid Community Group](https://www.w3.org/community/solid/charter/) will be used. If there are three or fewer self-nominees, they will become the Operations Advisory Group without an election.

Author: @csarven @pchampin do you know if it is possible to re-use existing W3C voting infrastructure to do this?

If so could you help me do this - we would like to form this operations advisory committee as early as possible given that there are some decisions we would like their input on in the near future.

Reviewer 2: Pierre-Antoine can answer more definitely but I highly doubt it can be re-used. Solid ODI Governance falls outside the W3C gov/space. [Solid CG used W3C's WBS for CG chair voting](https://github.com/solid/specification/discussions/582), and that was under the mutual understanding that it would be an experiment - whether it can be re-used or officially available for all CGs remains is not yet established. If you want a more authoritative/direct answer, best to contact `Team Community Process <team-community-process@w3.org>`.

IIRC, polls can be set-up with the W3C system, but that's different than voting, and it may also be only for WGs. However, that's still for W3C/CG stuff, and as I understand it, Solid ODI Governance is a separate matter. I'm also not sure I understand the relevance of the Solid Operations Advisory Group for the W3C Solid CG or the Solid CG charter or Solid CG participants. (I understand it'd be super easy if this was possible but Solid ODI may have to find other ways to admin / organise stuff..)

Author: Hi Sarven,

Thanks for the infrastructure info! I was sort of hoping the software infrastructure for voting was open source, and we could use it to run the vote for the Operations Advisory Group. I'll email the contact you listed.

Indeed, there is not much relevance to the charter but was expecting those who want to input on infrastructure to be an approximate subset of Community Group; and affiliations are already declared there which helps identify when multiple votes are coming from the same org (which iirc is accounted for and deduped).

So the CG is largely just to identify the pool of people who can vote. Do you have an alternate proposal for how to select this pool of people if you think this is an inappropriate selection mechanism.

Reviewer 2: I hear you completely. And, you're going through a similar process as I've with respect to determining eligible voters and taking some constraints into account. The difference is that we were able to lean on W3C a bit - most importantly authenticating the eligible voters and the voting itself - but your situation is more complicated as I understand it.

I think it is best to not limit (or otherwise constrain) participation in the Solid community to W3C Solid CG, you'll have to work through an alternative. Right that it'd be an inappropriate selection mechanism.

One thing that immediately comes up is whether there should be full transparency in votes as far as identifying the voters or even the candidates and how they rank. I suspect that some consensus can be reached on that but it won't be unanimous, i.e., some people will, for the many reasons, prefer to maintain their anonymity. And, I think you should try to maintain that possibility/requirement at first. For instance, while GitHub could be of use, ticks the authentication box where you can also cross-check with eligible voters, but voters/candidates will be identifiable.

At the early states of writing the chair selection procedure, I've briefly considered to use a third-party service for voting and whatnot (because W3C WBS wasn't available to CGs) but quickly wanted to move away from that whole idea because it was going to be a lot more complicated than using the W3C system (easy to work CG participants where they can authenticate any way). Getting people to create an account at an arbitrary voting service, where you can verify their eligibility, and then dealing with voting is going to be PITA. So, this should really be your last resort IMO.

https://github.com/w3c/wbs-design is not particularly what you're looking for but perhaps the W3C Team can provide you with better pointers.

Reviewer 4: +1 to all @csarven wrote above.

The way I see it: if ODI wants to equate "CG participants" and "eligible voters", then somehow it becomes a CG vote -- if the CG chairs agree to that. Then on behalf of the CG, the CG chairs could ask team-community-process@w3.org to use WBS again.

Author: Thanks both for the pointers! 

Have removed the CG participants = eligible voters requirement (https://github.com/solid/odi-governance/pull/20#discussion_r2003950124).
